developer_manual/app/tutorial.rst

```diff
@@ -0,0 +1,487 @@
+                </field>
+            </declaration>
+        </table>
+    </database>
+
+To create the tables in the database, the version tag in **ownnotes/appinfo/info.xml** needs to be increased:
+
+.. code-block:: xml
+
+    <?xml version="1.0"?>
+    <info>
+        <id>ownnotes</id>
+        <name>Own Notes</name>
+        <description>My first ownCloud app</description>
+        <licence>AGPL</licence>
+        <author>Your Name</author>
+        <version>0.0.2</version>
+        <namespace>OwnNotes</namespace>
+        <category>other</category>
+        <dependencies>
+            <owncloud min-version="8" />
+        </dependencies>
+    </info>
+
+Reload the page to trigger the database migration.
+
+Now that the tables are created we want to map the database result to a PHP object to be able to control data. First create an :doc:`entity <database>` in **ownnotes/db/note.php**:
+
+
+.. code-block:: php
+
+    <?php
+    namespace OCA\OwnNotes\Db;
+
+    class Note extends \OCP\AppFramework\Db\Entity implements JsonSerializable {
+
+        protected $title;
+        protected $content;
+        protected $userId;
+
+        public function jsonSerialize() {
+            return [
+                'id' => $this->id,
+                'title' => $this->title,
+                'content' => $this->content
+            ];
+        }
+    }
+
+.. note:: A field **id** is automatically set in the Entity baseclass
+
+We also define a jsonSerializable method and implement the interface to be able to transform the entity to JSON easily.
+
+Entities are returned from so called :doc:`Mappers <database>`. Let's create one in **ownnotes/db/notemapper.php** and add a **find** and **findAll** method:
+
+.. code-block:: php
+
+    <?php
+    namespace OCA\OwnNotes\Db;
+
+    use \OCP\IDb;
+
+    class NoteMapper extends \OCP\AppFramework\Db\Mapper {
+
+        public function __construct(IDb $db) {
+            parent::__construct($db, 'ownnotes_notes', '\OCA\OwnNotes\Db\Note');
+        }
+
+        public function find($id, $userId) {
+            $sql = 'SELECT * FROM *PREFIX*ownnotes_notes WHERE id = ? AND user_id = ?';
+            return $this->findEntity($sql, [$id, $userId]);
+        }
+
+        public function findAll($userId) {
+            $sql = 'SELECT * FROM *PREFIX*ownnotes_notes WHERE user_id = ?';
+            return $this->findEntities($sql, [$userId]);
+        }
+
+    }
+
+.. note:: The first parent constructor is the database layer, the second one the table that is used and the third is the entity on which the result should be mapped onto. Insert, delete and update methods are already implemented.
+
+Connect Database & Controllers
+==============================
+The mapper which provides the database access is finished and can be passed into the controller.
+
+In general it is good practice to use another class between mappers and controllers called services, but for the sake of brevity we will leave them out.
```

Comment:
A short sentence additional sentence describing what a service actually is good for would make sense regarding the audience.